### ISSUE TITLE
Change Vertical Scrollbar Width

### ISSUE THREAD

**Issue by cocowalla** [microsoft/vscode#30191]
Is there any way to change the width of the vertical scrollbars in Visual Studio Code, especially the one for the left panel?

I find it so thin that it's a bit fiddly to use.

**Comment by alexandrudima** [comment]
I don't think we currently have any option to customize the width of the vertical scrollbars in the viewlets.

**Comment by cocowalla** [comment]
Is this an Electron limitation? Or do you mean that a means simply doesn't exist in VS Code yet?

**Comment by alexandrudima** [comment]
The scrollbars are all custom. e.g. you can change the width of the scrollbars instances in the editor via a hidden option e.g. `"editor.scrollbar.verticalScrollbarSize": 50` (requires a F1 > Reload window / VSCode restart to apply)

**Comment by cocowalla** [comment]
@alexandrudima I get a squiggly line under `"editor.scrollbar.verticalScrollbarSize"` with a tooltip that says `Unknown configuration setting` - but it does indeed work, thanks!

**Comment by nicooprat** [comment]
It works indeed. Tried the same for the file explorer in the sidebar, but `"explorer.scrollbar.verticalScrollbarSize": 5` doesn't work though. Any other hidden feature? :)

**Comment by jez9999** [comment]
This gets my vote as well.  Many times now I have wanted that explorer scrollbar to be thicker.  I am not on a mobile device!!

**Comment by disrupted** [comment]
does someone know how to get rid of this atrocious "Unknown configuration setting" warning?

**Comment by paterasMSFT** [comment]
Why was this closed? explorer.scrollbar.verticalScrollbarSize still doesn't work, and editor.scrollbar.verticalScrollbarSize still has an unknown setting warning (though it does work).

@aeschli, you reference #24755, which you've closed, saying an issue about scrollbar width should be opened. But you've also closed this issue, which is about scrollbar width. You even referencing this issue when you say an issue should be opened.

What was your intention?

**Comment by aeschli** [comment]
@paterasMSFT This issue is still open. #24755 was about to issues: color and width. For colors we added some new color ids. Width is covered by this issue.

**Comment by paterasMSFT** [comment]
Oh, sorry, I thought the Closed label after the reference to #24755 referred to you closing this issue.

**Comment by AndreSouzaAbreu** [comment]
Hello, everyone!

My case is a little bit different and I'm attaching a screen picture to show what I mean. Could you help with that?
I want to change the thickness of the horizontal scrollbar on the TOP. This scrollbar is the one that allows you to switch between files. I drew a green square to indicate the scrollbar I want to change.

By the way, following the instructions here, I put the code in the settings.json (as you can see inside the red square) and it changed the horizontal scrollbar on the bottom (which is indicated by a red arrow). I do NOT need this one. 

How can I change the top scrollbar to make it thicker and always visible?

I hope I was clear. If not, please, let me know how I can improve my questions.

![vscode scrollbar](https://user-images.githubusercontent.com/47862859/53285743-f6d47a00-3742-11e9-99c2-411c188345e9.PNG)

**Comment by VipSaran** [comment]
Nice discussion with some workarounds, but I'm missing the one for the tab scroll bar, unfortunately. Is there a (hidden) option to change/increase its height? Because it's tricky to grab it an not to unintentionally move one of the tabs.

![image](https://user-images.githubusercontent.com/995948/57908941-aa9c6000-7881-11e9-9860-9df4d7ac07be.png)


Possibly it is handled via horizontalSliderSize?

**Comment by edmondburnett** [comment]
Another vote to have the option to change the size of the horizontal tab scrollbar. It's pretty difficult to grab and I end up dragging the tabs instead of the scrollbar around most of the time.

**Comment by jeffrey-hui** [comment]
I haven't noti

> Nice discussion with some workarounds, but I'm missing the one for the **tab scroll bar**, unfortunately. Is there a (hidden) option to change/increase its height? Because it's tricky to grab it an not to unintentionally move one of the tabs.
> 
> ![image](https://user-images.githubusercontent.com/995948/57908941-aa9c6000-7881-11e9-9860-9df4d7ac07be.png)
> 
> Possibly it is handled via horizontalSliderSize?

I didn't notice a difference for the tab scroll bar when changing the `"editor.scrollbar.horizontalScrollbarSize"`, is there another setting to change the tab scroll bar only?

**Comment by tiotolstoy** [comment]
> It works indeed. Tried the same for the file explorer in the sidebar, but `"explorer.scrollbar.verticalScrollbarSize": 5` doesn't work though. Any other hidden feature? :)

Did anyone ever figure this out? Explorer sidebar is really bugging me since I have a tiny computer.

**Comment by musicdsign** [comment]
> Hello, everyone!
> 
> My case is a little bit different and I'm attaching a screen picture to show what I mean. Could you help with that?
> I want to change the thickness of the horizontal scrollbar on the TOP. This scrollbar is the one that allows you to switch between files. I drew a green square to indicate the scrollbar I want to change.
> 
> By the way, following the instructions here, I put the code in the settings.json (as you can see inside the red square) and it changed the horizontal scrollbar on the bottom (which is indicated by a red arrow). I do NOT need this one.
> 
> How can I change the top scrollbar to make it thicker and always visible?
> 
> I hope I was clear. If not, please, let me know how I can improve my questions.
> 
> ![vscode scrollbar](https://user-images.githubusercontent.com/47862859/53285743-f6d47a00-3742-11e9-99c2-411c188345e9.PNG)

Yes I am having the same issue. I try to scroll at the top but select a file instead because the scrollbar is not tall enough.

**Comment by AndreSouzaAbreu** [comment]
With the release of Vscode 1.44.1, we finally have the option to configure the thickness of the top scrollbar :D ! I'm so glad to share this with you all !


![Screenshot](https://user-images.githubusercontent.com/47862859/79384484-57bda380-7f3d-11ea-94c9-0f31e111a815.png)

**Comment by rauschbit** [comment]
We've waited for this for a long time and I'm happy that there finally is an option for it and usually don't want to complain on that stuff - but why is there only the option "large" instead of giving the users the choice to set their own thickness like for many other sizing options?

**Comment by AndreSouzaAbreu** [comment]
> We've waited for this for a long time and I'm happy that there finally is an option for it and usually don't want to complain on that stuff - but why is there only the option "large" instead of giving the users the choice to set their own thickness like for many other sizing options?

I'm sorry, but I don't know. I'm just sharing the information to make people aware of it. Well, at least we have a partial solution to the problem

**Comment by VipSaran** [comment]
Even though it's implemented, there's (at least one) bug for it already:
https://github.com/microsoft/vscode/issues/95413

Vote to give it exposure ;)

**Comment by kuhlaid** [comment]
What about changing the terminal window scrollbar? I tried
`"terminal.scrollbar.verticalScrollbarSize": 50` with no luck. This scrollbar is painfully tiny.

**Comment by nathan-alden-sr** [comment]
To VS Code maintainers: Come on, folks, it should be very, very obvious that enumeration-based settings, or only providing a setting for _one_ visual element's scrollbar is _not_ what is being asked for here. I want the ability--and I suspect most people are the same--to customize _all_ scrollbars' widths/heights. For something so basic I am shocked this hasn't been implemented, especially considering how otherwise flexible VS Code is in almost every regard.

**Comment by kuhlaid** [comment]
I'm not sure which recent code version increased the scrollbar width of the terminal pane but thank you. It is much easier to attach my cursor to the scrollbar now.

**Comment by jez9999** [comment]
Both horizontal and vertical scrollbars in the code editor, as well as the vertical scrollbar in the Explorer, should have settings allowing the user to specify how thick they should be. Setting should be a number, not values like "small" and "large".

**Comment by gjsjohnmurray** [comment]
@jez9999 even now you can achieve this for the editor by using the 'hidden' settings and reloading/restarting, e.g.

```
	"editor.scrollbar.horizontalScrollbarSize": 40,
	"editor.scrollbar.verticalScrollbarSize": 50,
```

I agree they should become official, and there should be equivalent settings for elsewhere (e.g. Explorer).

**Comment by Kenobi-the-2nd** [comment]
I agree. The width adjustment for the explorer scroll bar would be incredibly helpful.

**Comment by ypapouin** [comment]
Yes indeed, I don't have a scrollwheel on my pen, and on a daily usage, it
is a pain to scroll the explorer without starting dragging a file/folder.

**Comment by viponedream** [comment]
scroll bar in explore still too thin.  hard to drag.
![scroll bar too small](https://user-images.githubusercontent.com/16624529/85329140-87bc9080-b504-11ea-906a-2be919d8c379.png)

**Comment by diegoshe** [comment]
> Всем привет!
> 
> Мой случай немного отличается, и я прилагаю экранное изображение, чтобы показать, что я имею в виду. Не могли бы вы помочь с этим?
> Я хочу изменить толщину горизонтальной полосы прокрутки на вершине. Эта полоса прокрутки позволяет переключаться между файлами. Я нарисовал зеленый квадрат, чтобы указать полосу прокрутки, которую я хочу изменить.
> 
> Кстати, следуя приведенным здесь инструкциям, я поместил код в файл settings.json (как вы можете видеть внутри красного квадрата), и он изменил горизонтальную полосу прокрутки внизу (которая обозначена красной стрелкой). Мне не нужен этот.
> 
> Как изменить верхнюю полосу прокрутки, чтобы она стала толще и всегда была видна?
> 
> Надеюсь, мне было ясно. Если нет, пожалуйста, дайте мне знать, как я могу улучшить свои вопросы.
> 
> ![полоса прокрутки vscode](https://user-images.githubusercontent.com/47862859/53285743-f6d47a00-3742-11e9-99c2-411c188345e9.PNG)

Use
"workbench.editor.titleScrollbarSizing": "large",

**Comment by josehcif** [comment]
I am another VSCode user waiting for the explorer vertical scrollbar to be changed to wider setting. It is really annoying as is now, since you need to have superhuman precision to grab it.

**Comment by Christopher-James-Francis-Rodgers** [comment]
English translation of diegoshe's comment two posts above this post.
-------------
Hello!

My case is a little different and I am attaching a screen image to show what I mean. Could you help with this?
I want to change the thickness of the horizontal scrollbar at the top. This scroll bar allows you to switch between files. I drew a green square to indicate the scrollbar I want to change.

By the way, following the instructions given here, I put the code in the settings.json file (as you can see inside the red square) and it changed the horizontal scrollbar at the bottom (which is denoted by the red arrow). I don't need this one.

How can I change the top scrollbar to make it thicker and always visible?

I hope it was clear to me. If not, please let me know how I can improve my questions.

~diegoshe

**Comment by Kenobi-the-2nd** [comment]
Is it possible we could get this feature added at some point in the near future? I realize the VS code team stays busy working on improvements (which is much appreciated!), but this feature request was opened almost 3 years ago and has substantial backing. Adding this would definitely remove the frustration alot of us have with using the miniature scroll bar in the explorer window. 

Thank-you!

**Comment by alexdima** [comment]
Appologies, I might have closed this too quickly. `editor.scrollbar.vertical` is now a real setting, but that only covers the editor.

**Comment by katerlouis** [comment]
Any news here?

**Comment by dsuch** [comment]
Hello,

yes, it has taken me far too long to bring the normal scrollbacks back. 

On VS Code 1.62.3 (ccbaa2d27e38e5afa3e5c21c1c7bef4657064247), here is what I have so far. It would be great if someone could add all the other missing parts.

Note that in my case the panel is on the right-hand side.

![scrollbar](https://user-images.githubusercontent.com/553332/143673081-e09387dd-65d2-4e52-8b2d-b602316b1584.png)

Here are the changes:

* To display the scrollbar in the editor: add ```"editor.scrollbar.verticalScrollbarSize": 20,``` to user settings
* To display the scrollbar in the panel:
   * Add a custom CSS with [Custom CSS and JS Loader
](https://marketplace.visualstudio.com/items?itemName=be5invis.vscode-custom-css)
  * To that file, add this snippet:
```css
.monaco-scrollable-element>.visible {
    background: #f3f3f3;
    width: 20px !important;
}
```
* To permanently display the scrollbar in the panel, back up file **workbench.desktop.main.js** in your VS Code installation and delete this snippet ```&&(this._verticalScrollbar.beginHide(),this._horizontalScrollbar.beginHide())```

The missing and incomplete functionality is:

* No scrollbar handles
* In the panel scrollbar, the thumb is thinner than the background

Could someone be please so kind to learn how to add these missing pieces and post the resolution to this ticket? Hopefully this should be only a matter of custom CSS.

**A short note to developers** - there is no glory in adding features just for the sake of innovation.

At one point, you are overshooting your users and customers with changes that are not required and that are added purely because that can be done, not because they are needed.

Constantly tweaking with scrollbars, hiding, removing, fading out or obscuring them along with similar practices are the kind of low-hanging fruits that bring no progress. These are not problems that need resolving.

I realise that you are merely a part of a bigger trend among GUI designers who are not sure what to do next but understand that at one point the product, or a given part of it, is complete. The next step is to find a new product, not to find a way to constantly modify the existing one.

Resist the urge to modify just because it can be done. We already know that you are capable of building a good product. There is no shame in not changing what works well. Indeed, this is where the wisdom lies.

Thank you.

**Comment by inaryart** [comment]
any plan for all scrollbar

**Comment by logicbus** [comment]
I got pointed to this issue because I created a similar issue that didn't get enough votes: #152113 .

My first recollection of a disappearing scrollbar was the first iPhone.  On a smartphone screen, UI elements have to be pretty large in order to be interacted with.  So a scrollbar exists there more as a feedback mechanism -- an indicator of where you are in a page, and maybe how long the page is -- rather than a tool to convert user input.  

Many paradigms from small, touch-enabled screens have made their way to large, traditional screens, and some have been for the worse.  This is one of them.  Yes, showing the scrollbar all the time whether its needed or not is a waste of space. (Didn't IE11 do that?) And VS Code packs a lot in a single window.  But making scrollbars hard to interact with -- or hard to find! -- sucks.  

Make them easy to see, make them easy to drag, and maybe don't make them disappear after half a second.

**Comment by nathan-alden-sr** [comment]
Pretty crazy that after nearly four decades of visual interfaces in software, there is still software (Visual Studio Code) that doesn't even implement scrollbars well. C'mon, Microsoft. This has been a solved problem for _decades_.

**Comment by jez9999** [comment]
```
"editor.scrollbar.horizontalScrollbarSize": 40,
"editor.scrollbar.verticalScrollbarSize": 50,
```

I'd just note that the above doesn't work for the scrollbar in the explorer to the left.  Also it'd be nice to have a way to keep all scrollbars open (but I suspect that's hoping for too much; they'd also have to stop being iPhone-style "disappearing" scrollbars, which are obviously completely appropriate for developing code on a PC with a high resolution monitor).

**Comment by xXNickznXx** [comment]
I opened another issue on the same topic some time ago: https://github.com/microsoft/vscode/issues/160804

And now I see that there is this one. Mine kind of extends this one and I also see alot of comments here, that are on the same topic as mine, but I don't know how to deal with it now.

**Comment by avanember** [comment]
Why is so hard to implement this? It's very annoying.

**Comment by davidjoffe** [comment]
Adding a vote to please be able to (1) disable _scrollbar fading_, (2) disable _scrollbar disappearing_ (_except_ if content not scrollable, then it _should_ disappear), and (3) be able to increase/configure width of explorer scrollbar

There's a principle in User Interface design 'larger targets are easier to hit' ([Fitt's Law](https://en.wikipedia.org/wiki/Fitts%27s_law), a principle that some days it feels has become forgotten knowledge). Scrollbars generally used to be wider relative to display width, then at some stage there was a trend toward impractically narrower scrollbars

Scrollbars serve important _practical_ purposes to users even if some may not like them

This may sound like a minor nitpick, but I'm trying to do a tricky merge task of four different branches of a repo loaded at once so I can quickly jump around and view/compare/edit/consolidate the four different branches via explorer, and not being able to see the scroll position means I can't even see what repo I'm looking at (the most important thing). (Combine this with the overly narrow scrollbar on my high-DPI hi-res laptop - they're really very narrow - and distracting auto-fading). I'll try Visual Studio instead.

I know VS Code is free, so I'm not bashing and I don't expect devs should work for free if there's no time to implement this. If I had time I'd try submit a PR

(The problem is compounded by the resize grab-handle also being invisible until you float the mouse over it, something I'd also always prefer to always be visible), so it's easy to hit the resize grab when aiming for the scrollbar.)

Sorry for long comment

**Comment by dsuch** [comment]
@davidjoffe You should not be apologizing for clearly expressing your thoughts.

VS Code is a proprietary, commercial product from Microsoft, released as freeware. The reason why this functionality is not implemented is that owners of this product at Microsoft do not want for it to be implemented. This has nothing to do with "developers not having time" and we can assume that owners do not read comments here either.

I imagine that the only way to add what is obviously a needed feature is to "petition" Microsoft until someone high enough notices that or, more likely, find someone inside Microsoft who is a contact of a contact and have that inside person act as champion of this cause.

Basically, we need for this to become a pet project of one of the VPs. Thus, if you have such an insider among your contacts at Microsoft and you can explain this situation, that would be a better way to get this done.

**Comment by heinrich-ulbricht** [comment]
Another vote from my side. After increasing the sash width to something that can actually be used via RDP connection it nearly completely overlaps the scroll bar of the tool windows. Now I'd like to make the scroll bars wider as well, to be able to use them. And this brought me here ^^

**Comment by aifirstd3v** [comment]
There is no scrollbar setting option for Explorer, Sidebar.

**Comment by tylerlaprade** [comment]
I just want a single setting for all my horizontal scrollbars, and another for all my vertical scrollbars.

**Comment by starball5** [comment]
Related on Stack Overflow: [How to make scrollbars wider in Visual Studio Code?](https://stackoverflow.com/q/50106972/11107541)

**Comment by s-h-a-d-o-w** [comment]
It's bizarre the lengths to which many tech companies go for accessibility, yet the simplest and most basic thing of improving accessibility, they just can't seem to bring themselves to do.

**Comment by s-h-a-d-o-w** [comment]
I've come up with a workaround that works for almost all scrollbars by injecting CSS:
https://github.com/s-h-a-d-o-w/accessible-scrollbars?tab=readme-ov-file#vscode

(I attempt at what @davidjoffe suggested, always showing scrollbars. Unfortunately, that would also show scrollbars when there's nothing to scroll - i.e. no slider. At least as a default, I've found that to be too intrusive but you can uncomment it in the CSS file.)

**Comment by volkb79-2** [comment]
its been about a year and `editor.scrollbar.verticalScrollbarSize`  is still missing its companion Explorer /workbench tree/list widget setting ;(

often scrolling in long explorer trees really is hard hitting the tiny scrollbar :/
